example/tests/test_openapi.py

```diff
@@ -0,0 +1,168 @@
+# largely based on DRF's test_openapi
+import json
+
+from django.test import RequestFactory, override_settings
+from django.urls import re_path
+from rest_framework.request import Request
+
+from rest_framework_json_api.schemas.openapi import AutoSchema, SchemaGenerator
+
+from example import views
+from example.tests import TestBase
+
+
+def create_request(path):
+    factory = RequestFactory()
+    request = Request(factory.get(path))
+    return request
+
+
+def create_view_with_kw(view_cls, method, request, initkwargs):
+    generator = SchemaGenerator()
+    view = generator.create_view(view_cls.as_view(initkwargs), method, request)
+    return view
+
+
+def test_path_without_parameters(snapshot):
+    path = '/authors/'
+    method = 'GET'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'get': 'list'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_path_with_id_parameter(snapshot):
+    path = '/authors/{id}/'
+    method = 'GET'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'get': 'retrieve'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_post_request(snapshot):
+    method = 'POST'
+    path = '/authors/'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'post': 'create'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_patch_request(snapshot):
+    method = 'PATCH'
+    path = '/authors/{id}'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'patch': 'update'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_delete_request(snapshot):
+    method = 'DELETE'
+    path = '/authors/{id}'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'delete': 'delete'}
+    )
+    inspector = AutoSchema()
+    # DRF >=3.12 changes the capitalization of these method mappings which breaks the snapshot,
+    # so just override them to be consistent with >=3.12
+    inspector.method_mapping = {
```

Comment:
by only reading the comment is this still needed as we are now only supporting DRF 3.12+?